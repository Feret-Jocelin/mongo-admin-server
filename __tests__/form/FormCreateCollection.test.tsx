/* eslint-disable no-undef */
import React from 'react';
import { render, screen } from '@testing-library/react';
import { NextIntlClientProvider } from 'next-intl';
import { store } from '@/store/store';
import { Provider } from 'react-redux';
import FormCreateCollection from '@/app/[locale]/components/form/form-create-collection/FormCreateCollection';

jest.mock('next-intl/client', () => ({
    usePathname() {
      return '';
    },
    useRouter() {
      return {
        push: () => jest.fn(),
        replace: () => jest.fn(),
      };
    },
  }));

  const locale = 'fr';
  const message = require(`../../shared/locales/${locale}.json`)

  describe("FormCreateCollection", () => {
    it('renders', () => {
      const handleClose = jest.fn();
      const { container } = render(
        <Provider store={store}>
          <NextIntlClientProvider locale={locale} messages={message}>
            <FormCreateCollection 
              open={true}
              onClose={handleClose}
            />
          </NextIntlClientProvider>
        </Provider>
      );
      expect(container).toMatchSnapshot();
    })
  })