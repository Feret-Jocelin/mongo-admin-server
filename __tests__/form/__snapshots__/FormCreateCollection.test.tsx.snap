// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormCreateCollection renders 1`] = `
<div>
  <dialog
    class="modal"
  >
    <section
      class="modal__header"
    >
      <div
        class="modal__header--icon"
      >
        <div
          class="undefined pointer-cursor"
        />
      </div>
      <h2>
        Créer une collection
      </h2>
    </section>
    <section
      class="modal__container"
    >
      <form>
        <div
          class="input-container"
        >
          <label
            class="label"
          >
            Nom de la collection
          </label>
          <input
            class="input "
            placeholder="Nom de la collection"
            type="text"
            value=""
          />
        </div>
      </form>
    </section>
    <section
      class="modal__footer"
    >
      <button
        class="button full outline center"
        type="button"
      >
        Annuler
      </button>
      <button
        class="button full center"
        disabled=""
        type="submit"
      >
        Créer une collection
      </button>
    </section>
  </dialog>
</div>
`;
